pkg/iam/iam.go

```diff
@@ -142,3 +162,111 @@ func (c *IAMClient) OrganizationIAM(ctx context.Context, organizationID string)
 
 	return m, nil
 }
+
+// RemoveProjectIAM removes the given IAM policy membership.
+func (c *IAMClient) RemoveProjectIAM(ctx context.Context, iamMember *assetinventory.AssetIAM) error {
+	// The Golang API does not support IAM removal on projects yet.
+	args := []string{
+		"projects", "remove-iam-policy-binding",
+		iamMember.ResourceID,
+		fmt.Sprintf("--member=%s", iamMember.Member),
+		fmt.Sprintf("--role=%s", iamMember.Role),
+	}
+	if iamMember.Condition != nil {
+		args = append(args, fmt.Sprintf("--condition=%s", conditionString(*iamMember.Condition)))
+	}
+	if _, err := c.gcloud(ctx, args); err != nil {
+		return fmt.Errorf("failed to remove iam policy for project %s: %w", iamMember.ResourceID, err)
+	}
+
+	return nil
+}
+
+// RemoveFolderIAM removes the given IAM policy membership.
+func (c *IAMClient) RemoveFolderIAM(ctx context.Context, iamMember *assetinventory.AssetIAM) error {
+	// The Golang API does not support IAM removal on folders yet.
+	args := []string{
+		"resource-manager", "folders", "remove-iam-policy-binding",
+		iamMember.ResourceID,
+		fmt.Sprintf("--member=%s", iamMember.Member),
+		fmt.Sprintf("--role=%s", iamMember.Role),
+	}
+	if iamMember.Condition != nil {
+		args = append(args, fmt.Sprintf("--condition=%s", conditionString(*iamMember.Condition)))
+	}
+	if _, err := c.gcloud(ctx, args); err != nil {
+		return fmt.Errorf("failed to remove iam policy for folder %s: %w", iamMember.ResourceID, err)
+	}
+
+	return nil
+}
+
+// RemoveOrganizationIAM removes the given IAM policy membership.
+func (c *IAMClient) RemoveOrganizationIAM(ctx context.Context, iamMember *assetinventory.AssetIAM) error {
+	// The Golang API does not support IAM removal on organizations yet.
+	args := []string{
+		"organizations", "remove-iam-policy-binding",
+		iamMember.ResourceID,
+		fmt.Sprintf("--member=%s", iamMember.Member),
+		fmt.Sprintf("--role=%s", iamMember.Role),
+	}
+	if iamMember.Condition != nil {
+		args = append(args, fmt.Sprintf("--condition=%s", conditionString(*iamMember.Condition)))
+	}
+	if _, err := c.gcloud(ctx, args); err != nil {
+		return fmt.Errorf("failed to remove iam policy for organization %s: %w", iamMember.ResourceID, err)
+	}
+
+	return nil
+}
+
+// gcloud runs the gcloud command.
+func (c *IAMClient) gcloud(ctx context.Context, args []string) (int, error) {
```

Comment:
I'm really -1 on using gcloud for this. Installing and managing gcloud is insanely expensive, and the overhead of spawning a child process and managing the output is incredibly high. It's also an unnecessary layer of abstraction.

Reply:
Done